registry/storage/driver/gdrive/gdrive.go

```
@@ -0,0 +1,497 @@
+	r, _ := d.driveService.Files.List().Q(`"` + d.rootID + `" in parents and name = "` + path + `"`).
+		Fields("files(id, name, size, modifiedTime)").Do()
+	if r != nil && len(r.Files) > 0 {
+		return r.Files[0]
+	}
+	return nil
+}
+
+type bytesBuffer struct {
+	*bytes.Buffer
+}
+
+func (b *bytesBuffer) Close() (err error) {
+	return nil
+}
+
+// Reader retrieves an io.ReadCloser for the content stored at "path" with a
+// given byte offset.
+func (d *driver) Reader(ctx context.Context, path string, offset int64) (io.ReadCloser, error) {
+	file := d.getFile(path)
+	if file == nil {
+		return nil, storagedriver.PathNotFoundError{Path: path}
+	}
+	httpRes, _ := d.driveService.Files.Get(file.Id).Download()
+	byteData, _ := ioutil.ReadAll(httpRes.Body)
+	byteData = byteData[offset:]
+	return &bytesBuffer{bytes.NewBuffer(byteData)}, nil
+}
+
+func (d *driver) Writer(ctx context.Context, subPath string, append bool) (storagedriver.FileWriter, error) {
+	var size int
+	var bytes []byte
+
+	if append {
+		file := d.getFile(subPath)
+		if file != nil {
+			httpRes, err := d.driveService.Files.Get(file.Id).Download()
+			if err != nil {
+				log.Fatalf("Unable to retrieve the file in drive: %s", err)
+			}
+			newText, _ := ioutil.ReadAll(httpRes.Body)
+			size = len(newText)
+			bytes = newText
+		}
+
+	}
+	return newFileWriter(bytes, subPath, size, d), nil
+}
+
+// Stat retrieves the FileInfo for the given path, including the current size
+// in bytes and the creation time.
+func (d *driver) Stat(ctx context.Context, subPath string) (storagedriver.FileInfo, error) {
+	driveFileInfo := driveFileInfo{}
+	file := d.getFile(subPath)
+	if file != nil {
+		driveFileInfo.modifiedTime = file.ModifiedTime
+		driveFileInfo.size = file.Size
+		driveFileInfo.isDir = false
+		return fileInfo{
+			path:          subPath,
+			driveFileInfo: driveFileInfo,
+		}, nil
+	}
+
+	r, _ := d.driveService.Files.List().Q(`"` + d.rootID + `" in parents`).Do()
```

Don't ignore the error here.